framework/src/transfers/MultiAppMeshFunctionTransfer.C

```diff
@@ -219,7 +219,8 @@ MultiAppMeshFunctionTransfer::execute()
   {
     if (i_proc == processor_id())
       continue;
-    _communicator.send(i_proc, outgoing_points[i_proc]);
+    std::vector<Parallel::Request> reqs(n_processors());
```

Comment:
I don't think this is quite correct.  Your `reqs` vector is created and immediately goes out scope.  So there is no reason to have a vector of them, and also they never have `.wait()` called on them, which is not correct either.